### PR TITLE
Feature/fixed length passcode

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
-  Add fixed length validation example

Only thing that should be missing right now is the fade in animation for the checkmark.

### CODE REVIEW

**Review by ektasawant — COMMENTED** [review]
In mobile app bar height is 56px. Can we add ` height: 'calc(100vh - 56px)'` for mobile.

File: src/pages/form-validation/fixed-length-passcode/index.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                <div className={classes.container}>
+                    <Typography variant={'body1'}>
+                        Please enter the <strong>six-digit passcode</strong> we just send to you. The passcode is valid
+                        for 15 minutes.
+                    </Typography>
+                    <Typography variant={'body1'} style={{ marginTop: theme.spacing(1) }}>
+                        For the purpose of this demonstration, passcode <strong>123456</strong> will pass. Any other
+                        6-digit passcode will fail.
+                    </Typography>
+                    <Divider
```
Comment (ektasawant) [inline src/pages/form-validation/fixed-length-passcode/index.tsx:161]: Can we move it to style section?

**Comment by ektasawant** [comment]
Few comments:

1. After successful code entered we should disable field, currently it is enable. Can we disable it?
image

2. Enter wrong passcode > wait for error > start editing passcode again =  error message is still visible. Can we clear error when focus on input?

3. After incorrect passcode it shows an error with input selected
image

4. Background color for disable input field is not as per design may be need theme changes not sure :)
5. On hover input background color is different

**Comment by emclaug2** [comment]
@ektasawant  I'm starting to work through these changes but point #3, this is the intended behavior from figma: 
image

**Comment by emclaug2** [comment]
We have an issue already for the Selected Input bg color, it's here: https://github.com/pxblue/react-themes/issues/2
I'll check to see if the disabled filled input bg color is correct, if not I'll make an issue for this.

File: src/pages/form-validation/fixed-length-passcode/index.tsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    const [blurred, setBlurred] = useState(false);
+    const maxLength = 6;
+    const inputId = 'passcode-input';
+
+    setTimeout(() => {
+        const input = document.getElementById(inputId);
+        if (input) {
+            input.focus();
+        }
+    });
```
Comment (daileytj) [inline src/pages/form-validation/fixed-length-passcode/index.tsx:75]: Should be able to just use the useEffect hook with no dependencies. It is called after the first render so the input element should be there by the time this is called. 
```suggestion
    useEffect(() => {
        const input = document.getElementById(inputId);
        if (input) {
            input.focus();
        }
    }, [])
```

File: src/pages/form-validation/fixed-length-passcode/index.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            const currPasscode = event.target.value;
+            if (isNaN(currPasscode)) {
+                return;
+            }
+            setPasscode(currPasscode);
+            if (currPasscode.length === maxLength) {
+                onSubmit(currPasscode);
+            }
+        },
+        [passcode]
```
Comment (daileytj) [inline src/pages/form-validation/fixed-length-passcode/index.tsx:108]: You can get rid of this dependency, doesn't look like you're using it here.

**Review by huayunh — APPROVED** [review]
Looks good for the functionality and visual

**Comment by ektasawant** [comment]
> @ektasawant I'm starting to work through these changes but point #3, this is the intended behavior from figma:
> image

Oh my bad , In angular design pattern need to add this. Will log bug.

**Pull request merged**